examples/hitl/rearrange_v2/ui.py

```diff
@@ -633,6 +661,204 @@ def _update_hovered_object_ui(self):
+
+        if object_id is None:
+            return
+
+        if object_id is not None:
+            world = self._world
+            sim = self._sim
+            obj = sutils.get_obj_from_id(
+                sim, object_id, world._link_id_to_ao_map
+            )
+            if obj is not None:
+                # Get object category name.
+                object_category_name = world.get_category_from_handle(
+                    obj.handle
+                )
+                if object_category_name is None:
+                    object_category_name = "Object"
+
+                # Get object state controls.
+                object_state_controls = self._get_object_state_controls(
+                    obj.handle
+                )
+
+                # Get the primary region name.
+                primary_region = world.get_primary_object_region(obj)
+                if primary_region is not None:
+                    primary_region_name = primary_region.category.name()
+
+                # Get the contextual information.
+                color_ui_valid = [0.2, 1.0, 0.2, 1.0]
+                color_ui_invalid = [1.0, 0.2, 0.2, 1.0]
+                if self._is_object_pickable(object_id):
+                    if self._held_object_id == None:
+                        if self._world.is_any_agent_holding_object(object_id):
+                            contextual_info = (
+                                "The object is held by another agent."
+                            )
+                            contextual_color = color_ui_invalid
+                        else:
+                            if self._is_within_reach(obj.translation):
+                                contextual_info = "Double-click to pick up."
+                                contextual_color = color_ui_valid
+                            else:
+                                contextual_info = "Too far to pick up."
+                                contextual_color = color_ui_invalid
+                    elif self._held_object_id == object_id:
+                        if self._place_selection.point is not None:
+                            point = self._place_selection.point
+                            normal = self._place_selection.normal
+                            receptacle_object_id = (
+                                self._place_selection.object_id
+                            )
+                            placement_valid = (
+                                self._is_location_suitable_for_placement(
+                                    point, normal, receptacle_object_id
+                                )
+                            )
+                            if placement_valid:
+                                contextual_info = "Release to place."
+                                contextual_color = color_ui_valid
+                            else:
+                                contextual_info = "Cannot place object here."
```

Comment:
Yeah, this will be an injection point for more specific place constraint messages